### PR TITLE
Adds Knuth Morris Readme file

### PR DESCRIPTION
**Pull request by urbashi123**
Fixes #2555 

#### Checklist:
- [ ] 4 space indentation.
- [ ] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [ ] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- 
- 

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!
@SakshiShreya @asha15

### CODE REVIEW

File: Knuth_Morris_Pratt_Algorithm/KMP_README.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+﻿# KMP
```
Comment (raghav-dalmia) [inline Knuth_Morris_Pratt_Algorithm/KMP_README.md:1]: please avoid abbreviation

File: Knuth_Morris_Pratt_Algorithm/KMP_README.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+1. It was invented by Donald Knuth and Vaughan Pratt.
+2. It was the first linear time complexity algorithm for string matching which is used to find a pattern in a text.
+3. This algorithm compares character by character from left to right.
+
+**Components of KMP**
++ Prefix Table:In this algorithm whenever a mismatch occurs,it uses a preprocessed table known as Prefix Table to skip characters comparison while matching.
++ LPS Table:It stands for Longest proper Prefix which is also suffix.Sometimes prefix table is also known as LPS Table.
```
Comment (raghav-dalmia) [inline Knuth_Morris_Pratt_Algorithm/KMP_README.md:9]: please avoid abbreviation

File: Knuth_Morris_Pratt_Algorithm/KMP_README.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                           0123456
+Here all the characters of Patterns matched with a substring in Text which is starting from index value 15.
+So, we conclude that given Pattern found at index 15 in Text.
+
+
+
+##### TIME COMPLEXITY
+
+The time complexity of KMP  is O(n)
+
```
Comment (raghav-dalmia) [inline Knuth_Morris_Pratt_Algorithm/KMP_README.md:107]: also mention space complexity

File: Knuth_Morris_Pratt_Algorithm/KMP_README.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
```
Comment (raghav-dalmia) [inline Knuth_Morris_Pratt_Algorithm/KMP_README.md:108]: Add links of code from our repo

**Comment by raghav-dalmia** [comment]
Make sure to squash your commits after these changes.

**Comment by raghav-dalmia** [comment]
Squash your commits. I will merge it after that.

**Comment by urbashi123** [comment]
> Squash your commits. I will merge it after that.

I am unable to squash the commits.Can you help me?
@raghav-dalmia

**Comment by somya-kapoor** [comment]
For reference http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by somya-kapoor** [comment]
Due to no activity for more than 7 days, I am closing this PR

**Pull request closed without merging**